### ISSUE TITLE
Yum update is failing - Centos 6 EOL

### ISSUE THREAD

**Issue by malkam03** [phusion/holy-build-box#23]
Hello! The yum install/update commands stopped working for us recently. I suspect that this may be related to the [EOL of the Centos 6  base image](https://wiki.centos.org/About/Product) but I'm not sure.

The error is:
```bash
# yum update
Loaded plugins: fastestmirror, ovl
Setting up Update Process
Loading mirror speeds from cached hostfile
YumRepo Error: All mirror URLs are not using ftp, http[s] or file.
 Eg. Invalid release/repo/arch combination/
removing mirrorlist with no valid mirrors: /var/cache/yum/x86_64/6/base/mirrorlist.txt
Error: Cannot find a valid baseurl for repo: base
```

Is there any plan of upgrading the base image? Or maybe setting the [vault by default](https://www.getpagespeed.com/server-setup/how-to-fix-yum-after-centos-6-went-eol)?

**Comment by nightlark** [comment]
Thanks for the heads up!

The short answer right now is, I'm not sure. One of the main areas where the glibc used by CentOS 6 (or earlier) seems to still be common is Python wheels compliant with manylinux2010, so having a CentOS 6 image with a newer C++ compiler is still useful (there are already some issues with the 32-bit image using the default yum repository that will require building some dependencies from source to update it).

If switching over to the vault keeps install/update commands working as they are right now, then setting it as the default seems like a good intermediate step, maybe combined with another upgrade to the installed packages built from source before considering an upgrade to CentOS 7 for the base image.

**Comment by DenisSa** [comment]
I can confirm to have successfully switched over and installed a series of packages through the vault.

**Comment by faithanalog** [comment]
I've submitted a PR #24 that adds the `vault.` migration into the build process. looks to be working for me so far.

**Comment by FooBarWidget** [comment]
Fully addressed in #28. Releasing 2.0.2 now.
